Turn on macOS builds for Travis

brettcannon opened the pull request.

brettcannon committed:
Turn on macOS builds

brettcannon committed:
Temporarily disable change-type check to verify build under macOS

brettcannon committed:
Turn back on short-circuiting

matrixise commented:
Hi @brettcannon 

I think it can be more useful if you 'squash' your commits into one ?

brettcannon commented:
@matrixise why specifically? We squash at merge and so what are you hoping to gain if I do it just for the PR?

brettcannon committed:
Merge branch 'master' into test-mac

brettcannon committed:
Make mac passing optional until we know it's stable

brettcannon committed:
Fix how to delineate which builds are optional

Travis requires that all rules in allow_failures be matched for a build to be optional.

brettcannon committed:
Try yet again to make mac and coverage optional

brettcannon committed:
Travis only allows a single environment variable to be specified in the matrix
